shared/fileutil/fileutil.go

```diff
@@ -139,3 +144,107 @@ func CopyFile(src, dst string) error {
 	}
 	return nil
 }
+
+// CopyDir copies contents of one directory into another, recursively.
+func CopyDir(src, dst string) error {
+	dstExists, err := HasDir(dst)
+	if err != nil {
+		return err
+	}
+	if dstExists {
+		return errors.New("destination directory already exists")
+	}
+	if err := MkdirAll(dst); err != nil {
+		return errors.Wrapf(err, "error creating directory: %s", dst)
+	}
+	fds, err := ioutil.ReadDir(src)
```

Comment:
it will be better to read before `MkdirAll`. if directory is missing new dst dir will be made and fail in `dstExists` next time

Reply:
Yep, nice suggestion, will update the code now.

Reply:
Updated!